server.js

```diff
@@ -55,7 +56,11 @@ function Server (options) {
   var app = this
 
   function onError (e) {
-    app.reporter('runtimeError', app, undefined, e)
+    try {
+      process.stderr.write(errorHelper(e, app))
+    } catch (err) {
+      app.reporter('runtimeError', app, undefined, err)
+    }
```

Comment:
You misunderstand me. `onError` should contains only `app.reporter('runtimeError', app, undefined, e)`.

I told you to write try-catch in `listen()` method:

```
listen: function listen () {
  try {
    return BaseServer.prototype.listen.apply(this, arguments)
  } catch (e) {
    process.stderr.write(errorHelper(e, app))
  }
}
```